controller/src/main/java/io/pravega/controller/store/checkpoint/ZKCheckpointStore.java

```diff
@@ -80,6 +80,8 @@ public void setPosition(String process, String readerGroup, String readerId, Pos
     public Map<String, Position> getPositions(String process, String readerGroup) throws CheckpointStoreException {
         Map<String, Position> map = new HashMap<>();
         String path = getReaderGroupPath(process, readerGroup);
+        ReaderGroupData rgData = groupDataSerializer.deserialize(ByteBuffer.wrap(getData(path)));
+        rgData.getReaderIds().forEach(x -> map.put(x, null));
         for (String child : getChildren(path)) {
             Position position = null;
             byte[] data = getData(path + "/" + child);
```

Comment:
I think we should add a precondition check for the return value of `map.put(` at line 91 (below). If the put operation returns a null the check should fail ...